src/core/ledger/ledger.test.js

```diff
@@ -432,13 +437,31 @@ describe("core/ledger/ledger", () => {
     describe("mergeIdentitiy", () => {
       it("gives the target's grain balance and paid to the base account", () => {
         const ledger = ledgerWithActiveIdentities();
-        ledger._allocateGrain(id1, g("100"));
-        ledger._allocateGrain(id2, g("10"));
+        ledger._allocateGrain({id: id1, amount: g("100")}, allocationId1, 0);
+        ledger._allocateGrain({id: id1, amount: g("1")}, allocationId2, 2);
+        ledger._allocateGrain({id: id2, amount: g("10")}, allocationId3, 1);
         ledger.mergeIdentities({base: id1, target: id2});
 
         const account = ledger.account(id1);
-        expect(account.balance).toEqual(g("110"));
-        expect(account.paid).toEqual(g("110"));
+        expect(account.balance).toEqual(g("111"));
+        expect(account.paid).toEqual(g("111"));
+        expect(account.allocationHistory).toEqual([
+          {
+            grainReceipt: {id: id1, amount: g("100")},
+            timestampMs: 0,
+            allocationId: allocationId1,
+          },
+          {
+            grainReceipt: {id: id2, amount: g("10")},
+            timestampMs: 1,
+            allocationId: allocationId3,
+          },
+          {
+            grainReceipt: {id: id1, amount: g("1")},
+            timestampMs: 2,
+            allocationId: allocationId2,
+          },
+        ]);
```

Comment:
the test is titled `"gives the target's grain balance and paid to the base account"` and now its' actually testing extra properties about the allocation history and the sort order. this feels like potential confusion, e.g. what if someone re-writing this test doesn't realize that the timestamps were carefully chosen to require merge sorting and refactors in a way that stops testing that property. please write new test that is explicitly focused on the properties we're testing.

Reply:
accepted